domain/secret/state/state.go

```diff
@@ -3476,6 +3499,75 @@ GROUP BY sro.secret_id`
 	return result, nil
 }
 
+// ChangeSecretBackend changes the secret backend for the specified secret.
+func (st State) ChangeSecretBackend(
+	ctx context.Context, uri *coresecrets.URI, revision int,
+	valueRef *coresecrets.ValueRef, data coresecrets.SecretData,
+) error {
+	if valueRef != nil && len(data) > 0 {
+		return errors.New("both valueRef and data cannot be set")
+	}
+	if valueRef == nil && len(data) == 0 {
+		return errors.New("either valueRef or data must be set")
+	}
+	db, err := st.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	revSelectInput := secretRevision{
+		SecretID: uri.ID,
+		Revision: revision,
+	}
+	revSelectResult := revisionUUID{}
+	revSelectQ, err := st.Prepare(`
+SELECT uuid AS &revisionUUID.uuid
+FROM   secret_revision
+WHERE  secret_id = $secretRevision.secret_id
+       AND revision = $secretRevision.revision`, revSelectInput, revSelectResult)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	deleteValueRefQ, err := st.Prepare(`
+DELETE FROM secret_value_ref
+WHERE revision_uuid = $revisionUUID.uuid`, revisionUUID{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	deleteDataQ, err := st.Prepare(`
+DELETE FROM secret_content
+WHERE revision_uuid = $secretContent.revision_uuid`, secretContent{})
```

Comment:
We just use revision UUID so we can just use revSelectResult here too, no need for `secretContent{}`